charts/linkerd2-cni/values.yaml

```diff
@@ -26,8 +26,11 @@ destCNINetDir:    "/etc/cni/net.d"
 destCNIBinDir:    "/opt/cni/bin"
 # -- Configures the CNI plugin to use the -w flag for the iptables command
 useWaitFlag:      false
-# -- Kubernetes priorityClassName for the CNI plugin's Pods
-priorityClassName: ""
+# -- Kubernetes priorityClassName for the CNI plugin's Pods.
+# Defaults to system-cluster-critical so it signals the scheduler to start
+# before application pods, but after CNI plugins (whose priorityClassName is
+# system-node-critical). This isn't strictly enforced.
+priorityClassName: "system-cluster-critical"
```

Comment:
Is this needed for the cni-repair controller specifically? If not, can you pull it out into a separate change?

If CNI plugins should run at system-node-critical, why wouldn't the Linkerd CNI run at system-node-critical? If we don't have that as a default now, is there a reason for that? I.e. are there any downsides to setting this as a default?

If we omit this change from this PR, this change feels less risky to me.

Reply:
It's not required for this PR. The reasoning for using `system-cluster-critical` was to allow for the main CNI plugin in the cluster to run first, lessening the chance to run into the race condition the repair controller attempts to fix. But it appears these class names are either best-effort or the prioritization mechanism is simply not implemented as advertised, according to my testing. I'll remove this for now.